src/test/common/commandExecutor.test.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {CommandExecutor} from "../../common/commandExecutor";
+import {Log} from "../../common/log/log";
+
+import * as assert from "assert";
+import * as semver from "semver";
+import * as sinon from "sinon";
+
+suite("commandExecutor", function() {
+    suite("commonContext", function() {
+        teardown(function() {
+            let mockedMethods = [Log.logMessage, Log.logCommandStatus];
+
+            mockedMethods.forEach((method) => {
+                if (method.hasOwnProperty("restore")) {
+                    (<any>method).restore();
+                }
+            });
+        });
+
+        test("should execute a command", function() {
+            let ce = new CommandExecutor();
+            let actualExecuteResult: string[] = [];
+
+            sinon.stub(Log, "logMessage", function(message: string) {
```

Comment:
Because if we then change the signature of logMessage from: (message: string, formatMessage: boolean = true) to (formatMessage: boolean, message: string) or we make any kind of refactoring without introducing any bugs, the test using MemoryLogger would correctly pass, yet the one using stub.calledWith would incorrectly fail.